src/AudioTrackManager.ts

```diff
@@ -0,0 +1,93 @@
+import { AudioTrack } from "./types";
+import MediaAudio from "./MediaAudio";
+
+class AudioTrackManager {
+  public isPlaying: boolean;
+
+  private currentAudio: MediaAudio[];
+
+  constructor() {
+    this.isPlaying = false;
+    this.currentAudio = [];
+  }
+
+  public play(audioTracks: AudioTrack[]): void {
+    if (audioTracks.length === 0) return;
+
+    this.isPlaying = true;
+
+    audioTracks.forEach((audioTrack) => {
+      if (!audioTrack.url) return;
+
+      // Get the index of this current audio track if it exists.
+      const index = this.currentAudio.findIndex(
+        (currentAudio) => currentAudio.url === audioTrack.url,
```

Comment:
currentAudio never has items removed so this logic will probably not work as planned if I want to use example.com/office-background.mp3 in scene 1 and again in scene 3?

Reply:
This should remove the tracks when they end
```
        audio.onended = () => {
          this.currentAudio = this.currentAudio.filter(
            (currentAudio) => currentAudio !== audio,
          );
        };
```